pkg/pubsub/pubsub.go

```diff
@@ -0,0 +1,18 @@
+// Package pubsub provides an interface and implementations for publishing
+// notifications for Rekor updates to a Pub/Sub system.
+package pubsub
+
+import (
+	"context"
+
+	"github.com/sigstore/rekor/pkg/generated/models"
+)
+
+// Publisher provides methods for publishing notifications of new entries in the
+// transparency log.
+type Publisher interface {
+	// Publish publishes a transparency log entry to the configured Pub/Sub topic.
+	Publish(ctx context.Context, entry models.LogEntryAnon) error
```

Comment:
I'd like to see the entryID passed down here and published as a field within the pubsub message

Reply:
What consumer do you envision using this data, and for what purpose?

Reply:
the entryID uniquely identifies the entry within a given log (aka its merkle leaf hash). It can be recomputed by the other data in the bundle, but since we know it at the time the message is created it would be nice to pass it along as-is. 

We use it in the REST api to fetch a specific entry, e.g. `https://rekor.sigstore.dev/api/v1/log/entries/{entryID}`

Reply:
+1 entryID is going to be useful.

Reply:
(excuse my ignorance, still fairly new to the Rekor internals) What is the value in returning the `entryID` when the `LogIndex` is already returned as a part of `pb.TransparencyLogEntry`? There is an `/api/v1/log/entries?logIndex=<IDX>` which returns the same `'#/definitions/LogEntry'` according to the OpenAPI spec. I have confirmed this is handled using and `GetLogEntryByUUIDHandler` and `GetLogEntryByIndexHandler` respectively in `pkg/api/entries.go`.

Reply:
The `LogIndex` value does identify the entry and is used when computing an inclusion proof (we need to know whether there are 0 or 1 siblings of the leaf when computing the proof). Returning the entryID (which contains a shard identifier and the merkle leaf hash) could be considered an optimization, but is useful when verifying other calculations for definitively knowing which shard an entry is in, and ensuring the merkle leaf hash computation matches what is stored by the server.

Reply:
Thanks for the explanation. I've included entryID and it is used as the unique `id` in the CloudEvents field.